src/utils/convertData.js

```diff
@@ -0,0 +1,54 @@
+import dayjs from 'dayjs';
+import isBetween from 'dayjs/plugin/isBetween';
+import config from '../constants/config';
+
+const { days } = config;
+dayjs.extend(isBetween);
+
+const getTime = (date) => {
+  const time = date.set('m', 0);
+  return time.format('H');
+};
+
+const groupByDay = (posts) => {
+  const results = {};
+  // eslint-disable-next-line no-restricted-syntax
+  for (const post of posts) {
+    const date = dayjs.unix(post.created_utc);
+    const day = days[date.get('d')];
+    const time = getTime(date);
+
+    if (!results[day] || !results[day][time]) {
+      results[day] = { ...results[day], [time]: [] };
+    }
+
+    results[day][time].push(post);
+  }
+
+  return results;
+};
+
+const sort2 = (posts) => {
+  const sortedPosts = { ...posts };
+  Object.keys(sortedPosts).sort((a, b) => a - b);
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < 24; i++) {
+    if (!sortedPosts[i]) {
+      sortedPosts[i] = [];
```

Comment:
So sortedPosts is an object that looks like this, right?

```
{
  0: ...,
  1: ...,
}
```

In that case I would refactor this to be an array instead. You can access it the same way and your sorting will actually work

Reply:
yeah with all hours, and if there is  an hour missing I add an empty array